### DISCUSSION TITLE
Multiple OpenType features not working with DAveau font

### DISCUSSION THREAD

**Discussion by sid-wr** [harfbuzz/harfbuzz#4899] [General] [answered]
Hi,

There are multiple OpenType features like **_Swash_**, **_Stylistic Alternates_** etc not working with the DAveau font.

Not sure if the issue is at font level or HarfBuzz, as these OpenType features work for almost all the popular fonts.

I have logged 2 bugs for the issues which I found:
- https://github.com/harfbuzz/harfbuzz/issues/4897
- https://github.com/harfbuzz/harfbuzz/issues/4898

Can you please look into these ?

Thanks

**Comment by jfkthame** [comment, marked as answer]
Probably related to the presence of 'morx' as well as 'GSUB'/'GPOS'.

**Reply by khaledhosny** [reply]
Indeed, the font I found online has is shaped with the `morx`
```
$ hb-shape ~/Downloads/daveau/DaveauRegular.otf --features="ss01" --text="A" -V
trace: start table morx	buffer: [A=0]
trace: start chainsubtable 0	buffer: [A=0]
trace: end chainsubtable 0	buffer: [A=0]
trace: end table morx	buffer: [A=0]
trace: start table GPOS script tag 'latn'	buffer: [A=0+921]
trace: end table GPOS script tag 'latn'	buffer: [A=0+921]
[A=0+921]
```

The font appears to have been created by FontForge, so it is likely to be the `morx` table FontForge creates by “translating” GSUB lookups and is usually pretty useless or outright broken.
